src/dotnet/Telemetry.cs

```diff
@@ -19,6 +20,8 @@ public class Telemetry : ITelemetry
         private Dictionary<string, double> _commonMeasurements = null;
         private Task _trackEventTask = null;
 
+        internal static readonly string TelemetrySentinel  = Path.Combine(ApplicationEnvironment.ApplicationBasePath, $"{Product.Version}.dotnetTelemetry");
```

Comment:
@livarcocc took steps to guard against concurrent invocations despite it being a long operaiton. 

How does the Telemetry Sentinel differ in purpose form the nuget cache sentinel?